prelude/__internal__/Random.mad

```diff
@@ -1,456 +1,121 @@
+import type { Seed } from "LCG"
 import type { Maybe } from "Maybe"
+import LCG from "LCG"
 
-import Byte from "Byte"
-import ByteArray from "ByteArray"
-import Float from "Float"
-import Integer from "Integer"
-import List from "List"
-import { floor } from "Math"
-import Tuple from "Tuple"
 
 
+export type Seed
 
-// This code partially based upon: https://github.com/purescript/purescript-lcg/blob/master/src/LCG/LCG.purs
-// which has a BSD-3 License - https://github.com/purescript/purescript-lcg/blob/master/LICENSE
-
-// A seed for the linear congruential generator. We omit a `Semiring`
-// instance because there is no `zero` value, as 0 is not an acceptable
-// seed for the generator.
-export type Seed = Seed(Integer)
-
-
-/**
- * A SeededTuple is used to encapsulate a value relative to a seed
- * @since 0.22.0
- */
-export alias SeededTuple a = #[Seed, a]
-
-
-/**
- * The *modulus*: a magic constant for the linear congruential generator.
- * It is equal to 2^31 - 1, a Mersenne prime. It is useful for this value to
- * be prime, because then the requirement of the initial seed being coprime
- * to the modulus is satisfied when the seed is between 1 and LCG_M - 1.
- * @since 0.22.0
- */
-LCG_M :: Integer
-export LCG_M = 2147483647
-
-/**
- * The minimum permissible Seed value
- * @since 0.22.0
- */
-SEED_MIN :: Integer
-export SEED_MIN = 1
-
-/**
- * The maximum permissible Seed value
- * @since 0.22.0
- */
-SEED_MAX :: Integer
-export SEED_MAX = LCG_M - 1
-
-/**
- * The *multiplier*: a magic constant for the linear congruential generator
- * @since 0.22.0
- */
-LCG_A :: Integer
-export LCG_A = 48271
+export alias Random a = { get :: {} -> Seed, update :: {} -> {} }
 
 /**
- * The *increment*: a magic constant for the linear congruential generator
- * @since 0.22.0
- */
-LCG_C :: Integer
-export LCG_C = 0
-
-/**
- * This is the core logic for the "perturb" function
- * It is fundamentally applying (b * n + a) % d
- * @since 0.22.0
- */
-perturbation :: Integer -> Integer -> Integer -> Integer -> Integer
-export perturbation = (n, b, a, d) => Float.toInteger(
-    Integer.toFloat(b) * Integer.toFloat(n) + Integer.toFloat(a),
-  )
-    % d
-
-
-/**
- * Clamp a number between two boundaries, modularly
- * @since 0.22.0
+ * Create a Random instance from an LCG.Seed
+ * @since 0.23.0
  * @example
- * modClamp(5, 10, 100)
+ * gen = generateFromSeed(LCG.mkSeed(100))
  */
-modClamp :: Integer -> Integer -> Integer -> Integer
-export modClamp = (min, max, x) => {
-  y = x % (max - min)
-  return y < min ? y + max : y
+generateFromSeed :: Seed -> Random a
+export generateFromSeed = (initialSeed) => {
+  seed = initialSeed
+  get = () => seed
```

Comment:
Do we need get or could we just have update and update would then return the seed?

Reply:
I considered that, but I didn't want to cycle the seed too early (if for some reason you want to step manually -- probably small likelihood / use-case but I wanted to keep things explicit so that one could process things step-wise if needed)